tests/cpp17/generated_cpp17/monster_test_generated.h

```diff
@@ -878,6 +881,7 @@ inline flatbuffers::Offset<Stat> CreateStat(
 struct Stat::Traits {
   using type = Stat;
   static auto constexpr Create = CreateStat;
+  static const char* constexpr FileIdentifier = "MONS";
```

Comment:
I'm not sure how I feel about this, as a file identifier is typically associated with the root table type, not all of them. In your use case, how is `FileIdentifier` looked up on a non-root type? If that is not needed, should we leave this member out entirely for non-root types? Or make nullptr? or ""?

Also, `file_indentifier`, since this is not a type or function.

Reply:
I looked at generated code that has a `file_identifer` and code that doesn't have one. If it's not present, `nullptr` is passed. I think this makes sense to use.

Whether a type is a root type is orthogonal and should be IMO encoded differently:

```cpp
static constexpr bool root_type = ...
```

From a user perspective, it's important that all members in the traits are present, even if they don't apply. But you want to `if constexpr` with them to figure out which code path to choose.